drivers/digital-io/max22190/README.rst

```diff
@@ -0,0 +1,149 @@
+MAX22190 no-OS driver
+=====================
+
+Supported Devices
+-----------------
+
+`MAX22190 <https://www.analog.com/MAX22190>`_
+
+Overview
+--------
+
+The MAX22190 is an IEC 61131-2 compliant industrial digital input device.
+The MAX22190 translates eight, 24V current-sinking, industrial inputs to a 
+serialized SPI-compatible output that interfaces with 3V to 5.5V logic.
+A current setting resistor allows the MAX22190 to be configured for Type 1, 
+Type 2, or Type 3 inputs. Field wiring is verified for proximity switches, 
+by a second threshold detector on each input. When wire-break is enabled, 
+the active-low FAULT output is asserted and a register flag is set if the 
+input current drops below the wire-break threshold for more than 20ms. 
+Additional diagnostics that assert the active-ow FAULT pin include: 
+overtemperature, low 24V field supply, 24V field supply missing, 
+CRC communication error, etc.
+
+Applications
+------------
+
+MAX22190
+--------
+
+* Building Automation
+* Industrial Automation
+* Process Automation
+* Programmable Logic Controllers
+
+MAX22190 Device Configuration
+-----------------------------
+
+Driver Initialization
+---------------------
+
+In order to be able to use the device, you will have to provide the support
+for the communication protocol (SPI).
+
+The first API to be called is **max22190_init**. Make sure that it returns 0,
+which means that the driver was initialized correctly.
+
+Channel Configuration
+---------------------
+
+Each channel's state can be set with the **max22190_chan_state** API.
+
+Filter Configuration
+--------------------
+
+Filter can be bypassed for each channel separately, although if they are not,
+the user can set a filter delay.
+The user can also set a wire break detection for the filter.
+All this can be done with the **max22190_filter_set** API, and also the user
+can read the filter configuration for each channel with **max22190_filter_get**.
+
+MAX22190 Driver Initialization Example
+--------------------------------------
+
+.. code-block:: bash
+
+	struct max22190_desc *max22190_desc;
+	struct no_os_spi_init_param max22190_spi_ip = {
+		.device_id = 0,
+		.extra = &max22190_spi_extra,
+		.max_speed_hz = 100000,
+		.platform_ops = &max_spi_ops,
+		.chip_select = 1,
+	};
+	struct max22190_init_param max22190_ip = {
+		.comm_param = &max22190_spi_ip,
+		.crc_en = false,
+	};
+	ret = max22190_init(&max22190_desc, &max22190_ip);
+	if (ret)
+		goto error;
+
+MAX22190 no-OS IIO support
+--------------------------
+
+The MAX22190 IIO driver comes on top of the MAX22190 driver and offers support
+for interfacing IIO clients through libiio.
+
+MAX22190 IIO Device Configuration
+---------------------------------
+
+Device Attributes
+-----------------
+
+MAX22190 has a total of 6 attributes:
+* ``raw - the state of a channel``
```

Comment:
This doesn't render correctly, please test sphinx output

Also, you don't need to backtick quote. Perhaps use it on the attribute name to render like this (but in sphinx of course):
* `raw` - raw value of a channel

Reply:
will test the sphinx output, and make sure it renders correctly.